examples/psci/BasicEval.purs

```diff
@@ -0,0 +1,7 @@
+import Prelude
+import Data.Array
+
+let fac n = foldl mul 1 (1..n) in fac 10 -- 3628800
```

Comment:
It's to test the evaluated result of the let-in expr, `let fac n = foldl mul 1 (1..n) in fac 10`.